### DISCUSSION TITLE
SSH Tunnelling

### DISCUSSION THREAD

**Discussion by meglio** [pawelsalawa/sqlitestudio#4944] [General]
Appreciate your development effort. Any possibility to connect to a sqlite file that is accessible through an ssh tunnel remotely?

**Comment by pawelsalawa** [comment]
SQLite should not be used over any kind of network. It's not designed to work this way and any attempts to do it usually lead to transaction errors or data corruption. Some people use SQLite over NFS or Samba (Windows shared folders), but sooner or later they encounter unexpected and unexplained errors - this is because SQLite is designed and implemented to work locally. It's embedded database.

Since SQLite doesn't support and doesn't recommend it (https://www.sqlite.org/useovernet.html), so does the SQLiteStudio.

**Reply by pawelsalawa** [reply]
Note, that some people came here to report bugs, which actually turned out to be fault of using Samba shares for SQLite files.

**Reply by meglio** [reply]
@pawelsalawa  My use case is, I have my sqlite database in a mounted docker volume and it is used by docker containers orchestrated by docker compose. I would like to explore and manage the database from my host machine. Any ideas, or it's not a use case your app is seeking to support?

**Reply by pawelsalawa** [reply]
SSH adds networking layer to the I/O. This causes issues with SQLite itself. It doesn't matter what SQLiteStudio is seeking to support, because SQLite itself does not support (properly) such usage.

If you insist to use it anyway, just use sshfs to mount your SSH drive on your host system and then open it in SQLiteStudio just like any other local SQLite database. Just be warned - you may expect the unexpected, meaning errors here and there and it's not necessary SQLiteStudio or SQLite problem.

**Reply by meglio** [reply]
Thanks for the detailed explanation. Then for this particular use case it might worth trying a web-based application that I can install in one of the docker containers with the direct access to the file.
